test/test_ops.py

```diff
@@ -741,6 +741,18 @@ def test_backward(self, device, dtype, op):
             kwargs = sample.kwargs
             composite_compliance.check_backward_formula(op, args, kwargs)
 
+    @unittest.skipIf(IS_FBCODE or IS_SANDCASTLE, '__torch_dispatch__ does not work in fbcode')
+    @ops([op for op in op_db if op.supports_autograd], allowed_dtypes=(torch.float,))
```

Comment:
Don't check the "support_autograd" metadata, instead check if float is in the backward dtypes for the operator

Reply:
`op.supports_backward_dtype` requires `device` and we don't have `device` at this point.

Also I don't think there is an operator which doesn't support `torch.float` for backward 🤔 ?

Reply:
Why not check support_autograd?

Reply:
I'm going to remove it in favor of the check above because of patterns like this. In this case, for example, "supports_autograd" is metadata that is a proxy for "do you support autograd in float?", but that latter question is more precise and can be queried directly. "supports_autograd" makes it sounds like the operator supports autograd for all dtypes that can have grad, which is not always true.

Reply:
@mruberry can we query the backward dtype without knowing the device in this case?

Reply:
That's an excellent question: no. You have to move the filter into the test logic so you know which device it's instantiated for. Basically you just raise a skip from the test if the op doesn't support backward in float32.

Reply:
@mruberry can you please CC me when you're about to do that change? A lot of functorch testing rely on the supports_autograd / supports_forward_ad / etc flags right now

Reply:
Done.